build_tools/embed_data/build_defs.bzl

```
@@ -91,3 +91,86 @@ def cc_embed_data(
         testonly = testonly,
         **kwargs
     )
+
+def c_embed_data(
+        name,
+        srcs,
+        c_file_output,
+        h_file_output,
+        testonly = False,
+        strip_prefix = None,
+        flatten = False,
+        identifier = None,
+        **kwargs):
+    """Embeds 'srcs' into a C module.
+
+    Generates a header like:
+        #if __cplusplus
+        extern "C" {
+        #endif // __cplusplus
+        struct FileToc {
+          const char* name;             // the file's original name
+          const char* data;             // beginning of the file
+          size_t size;                  // length of the file
+        };
+        #if __cplusplus
+        }
+        #endif // __cplusplus
+
+        #if __cplusplus
+        extern "C" {
+        #endif // __cplusplus
+        const struct FileToc* this_rule_name__create();
+        #if __cplusplus
+        }
+        #endif // __cplusplus
+
+    The 'this_rule_name()' function will return an array of FileToc
+    structs terminated by one that has nullptr 'name' and 'data' fields.
```

`NULL` instead of `nullptr` for C?

```suggestion
    structs terminated by one that has NULL 'name' and 'data' fields.
```

---

Update. Thanks for the catch.